analyses/focal-cn-file-preparation/rna-expression-validation.R

```diff
@@ -0,0 +1,166 @@
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Read in data from tsv file (produced in `01-prepare-cn-file.R`)
+cn_df <- readr::read_tsv(file.path(results_dir, "annotated_cn.tsv"))
+
+# Read in RNA-seq expression data
+rsem_expression_polyA <-
+  readr::read_rds(file.path(root_dir,
+                            "data",
+                            "pbta-gene-expression-rsem-fpkm.polya.rds"))
+
+rsem_expression_stranded <-
+  readr::read_rds(file.path(
+    root_dir,
+    "data",
+    "pbta-gene-expression-rsem-fpkm.stranded.rds"
+  ))
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv"))
+
+#### Custom Function -----------------------------------------------------------
+
+merge_expression <-
+  function(copy_number_df, expression_df, metadata) {
+    # Given the focal copy number data.frame already annotated with the metadata,
+    # the RNA-seq expression data.frame, and the metadata, combine the data into one
+    # data.frame and plot expression values.
+    #
+    # Args:
+    #   copy_number_df: focal copy number data.frame annotated with the metadata
+    #   expression_df: RNA-seq expression data.frame
+    #   metadata: the relevant metadata data.frame
+    #
+    # Returns:
+    #   combined_df: data.frame with information from the focal CN, the
+    #                RNA-seq expression, and metadata data.frames
+    
+    # Change expression data.frame to long tidy format
+    long_expression <- expression_df %>%
+      tidyr::gather(biospecimen_id, expression_value,-gene_id) %>%
+      dplyr::distinct()
+    long_expression$gene_id <-
+      gsub(".*\\_", "", long_expression$gene_id)
+    
+    # Annotate expression data with metadata
+    expression_metadata <- long_expression %>%
+      dplyr::inner_join(metadata,
+                        by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+      dplyr::select(
+        biospecimen_id,
+        expression_value,
+        gene_id,
+        Kids_First_Participant_ID,
+        tumor_descriptor
+      ) %>%
+      dplyr::distinct()
+    
+    # Merge Focal CN data.frame with RNA expression data.frame
+    combined_df <- copy_number_df %>%
+      dplyr::distinct() %>%
+      dplyr::inner_join(
+        expression_metadata,
+        by = c(
+          "Kids_First_Participant_ID",
+          "gene_symbol" = "gene_id",
+          "tumor_descriptor"
+        )
+      )
+  }
+
+#### Join data -----------------------------------------------------------------
+
+# Add metadata to focal CN data.frame
+cn_df_metadata <- cn_df %>%
+  dplyr::inner_join(metadata,
+                    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::select(
+    gene_symbol,
+    biospecimen_id,
+    label,
+    copy_number,
+    Kids_First_Participant_ID,
+    tumor_descriptor,
+    tumor_ploidy
+  ) %>%
+  dplyr::filter(label %in% c("Hom_Deletion", "Hem_Deletion"))
```

Comment:
Move this to the first step (line 109). In general, you always want to filter before joining, as join gets more expensive the more rows it has to process.